libraries/lib-ipc/IPCClient.cpp

```diff
@@ -0,0 +1,77 @@
+#include "IPCClient.h"
+#include "IPCChannel.h"
+
+#include <thread>
+#include <atomic>
+#include <stdexcept>
+
+#include "internal/ipc-types.h"
+#include "internal/socket_guard.h"
+#include "internal/BufferedIPCChannel.h"
+
+class IPCClient::Impl
+{
+   std::atomic<bool> mAlive{true};
+   std::unique_ptr<std::thread> mWorker;
+public:
+
+   ~Impl()
+   {
+      mAlive = false;
+      if(mWorker)
+         mWorker->join();
+   }
+
+   void CreateChannel(int id, IPCChannelStatusCallback* callback)
+   {
+      if(mWorker)
+         throw std::runtime_error("cannot handle more than one channel");
```

Comment:
Return void and throw, or return boolean for error?  You made inconsistent choices for client and server.  And client, unlike server, makes no error if mImpl failed initialization on Windows.

I think it is eaiser to make both return boolean and not throw.